### PR TITLE
Add shellcheck action

### LINKED ISSUE #37: Enable shellcheck as github action
### Is your feature request related to a problem? Please describe.
Important Temporal scripts are in this repo now and `shellcheck` can help to find simple errors.

### PR DESCRIPTION
**Pull request by robholland**
Adds shellcheck workflow.

Fixes the one reported issue.

Fixes #37.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      - main
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Run ShellCheck
+        uses: ludeeus/action-shellcheck@master
```
Comment (mindaugasrukas) [inline .github/workflows/lint.yml:18]: I'm not sure about using the `master` ref here.

Also, could you give a context about why `ludeeus/action-shellcheck` over https://github.com/koalaman/shellcheck/wiki/GitHub-Actions?

Comment (robholland) [reply .github/workflows/lint.yml:18]: I took the action-shellcheck block from another repo, but I now can't find the repo I copied from... I'll switch to the canonical one.

Comment (robholland) [reply .github/workflows/lint.yml:18]: Hmm, the shellcheck action you linked to is for differential running. I don't think that's what we want here, I'd prefer all files in `docker` were always scanned. Also we only want to scan `./docker` as we have submodules with shell scripts which shouldn't be shellchecked as part of docker-builds CI. The action you linked to does not allow restricting paths (as far as I can see).

Comment (mindaugasrukas) [reply .github/workflows/lint.yml:18]: Actually, we don't want to run all repository files linting on a PR. The linter should lint only PR-affected changes.
The reason is that PR should have a single responsibility, but blocking it on unrelated issues is not a PR concern.

**Comment by robholland** [comment]
Updated.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (feedmeapples) [reply .github/workflows/lint.yml]: could be initially copied from here https://github.com/temporalio/cli/blob/1f078f2af258b5f6b1b14c7ef52d87a7f0e32905/.github/workflows/ci.yml#L33

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Run ShellCheck
+        uses: redhat-plumbers-in-action/differential-shellcheck@v4
+        with:
+          include-path: ./docker
```
Comment (mindaugasrukas) [inline .github/workflows/lint.yml:22]: nit: missing new line

**Review by mindaugasrukas — APPROVED** [review]
Approving this, to have any linter available. We can tune it later.

**Pull request merged**